### PR TITLE
Add release notes for PR #5781: bevy_reflect: Recursive registration

### LINKED ISSUE #1235: Write release notes for PR #12100: Add WinitEvent aggregate event for synchronized window event reading
https://github.com/bevyengine/bevy/pull/12100 needs release notes for the upcoming Bevy release!

Original authors: @UkoeHB

Please reply below if you'd like to write these notes. 
While the author(s) of the PR often have the context, knowledge and motivation to draft the release notes for their feature,
anyone can contribute release notes!

------

Release notes should:

1. Clearly motivate the change.
2. Be written in a way that is understandable by the average Bevy user: some programming background and a general understanding of games.
3. Show off the coolest features of the PR. Screenshots are awesome, but elegant APIs are also welcome!
4. If this was a perf-centric PR, quantify the performance improvements. Graphs and statistics work well for this.

We can help you revise the release notes: a rough draft alone is incredibly useful :)
Your expertise is invaluable for contextualizing the changes; we'll work with you to bring the technical writing up to par.

To submit your release notes, modify `..\release-content\0.14\release-notes\12100_Add_WinitEvent_aggregate_event_for_synchronized_window_eve.md` and submit a PR.
In that PR, please mention this issue with the `Fixes #ISSUE_NUMBER` keyphrase so it gets closed automatically.

### LINKED ISSUE #1236: Write release notes for PR #5781: bevy_reflect: Recursive registration
https://github.com/bevyengine/bevy/pull/5781 needs release notes for the upcoming Bevy release!

Original authors: @MrGVSV, @james7132, @soqb, @cart

Please reply below if you'd like to write these notes. 
While the author(s) of the PR often have the context, knowledge and motivation to draft the release notes for their feature,
anyone can contribute release notes!

------

Release notes should:

1. Clearly motivate the change.
2. Be written in a way that is understandable by the average Bevy user: some programming background and a general understanding of games.
3. Show off the coolest features of the PR. Screenshots are awesome, but elegant APIs are also welcome!
4. If this was a perf-centric PR, quantify the performance improvements. Graphs and statistics work well for this.

We can help you revise the release notes: a rough draft alone is incredibly useful :)
Your expertise is invaluable for contextualizing the changes; we'll work with you to bring the technical writing up to par.

To submit your release notes, modify `..\release-content\0.14\release-notes\5781_bevy_reflect_Recursive_registration.md` and submit a PR.
In that PR, please mention this issue with the `Fixes #ISSUE_NUMBER` keyphrase so it gets closed automatically.

### PR DESCRIPTION
**Pull request by MrGVSV**
Fixes #1236 

Adds release notes for recursive registration in `bevy_reflect`.

---

There's a small amount of nuance that I chose to gloss over such as why we're doing this instead of using `inventory` or `linkme`, or which specific things actually rely on these registrations.

I also chose not to go into detail about how this works under the hood via changes to `GetTypeRegistration` since most users probably don't even know that trait exists.

I did, however, mention that there is still an opportunity for a footgun if a type that's used on its own (i.e. a component, a resource, etc.) is implicitly registered by another type. This is hopefully a lot less likely to occur than forgetting to register an internal type, but still a possibility. This isn't required information, so we can remove it if its too much or detracts from the rest of the feature, but I thought it might still be good to draw attention to.

Anyways, let me know if there's anything that should be added/changed/removed!

### CODE REVIEW

**Comment by BD103** [comment]
Could you update the title `_release-notes.toml` to something like "Recursive type registration"? I think the "bevy_reflect" prefix it left over from the PR title.

**Review by BD103 — APPROVED** [review]
Reads well and is a good size, looks good! I have one nit about the title, but it's non-blocking. (We'll catch it later in the final pass.)

File: release-content/0.14/release-notes/5781_bevy_reflect_Recursive_registration.md [hunk 1]
```diff
@@ -1 +1,39 @@
-TODO
+Bevy uses reflection in order to dynamically process data for things like serialization and deserialization.
```
Comment (killercup) [inline release-content/0.14/release-notes/5781_bevy_reflect_Recursive_registration.md:1]:
```suggestion
Bevy uses [reflection](https://en.wikipedia.org/wiki/Reflective_programming) to dynamically process data for things like serialization and deserialization.
```

File: release-content/0.14/release-notes/5781_bevy_reflect_Recursive_registration.md [hunk 2]
```diff
@@ -1 +1,39 @@
...
+This often requires Bevy to know that a type even exists.
+This is why users need to register their types into the `TypeRegistry`.
```
Comment (killercup) [inline release-content/0.14/release-notes/5781_bevy_reflect_Recursive_registration.md:3]:
```suggestion
A Bevy app has a `TypeRegistry` to keep track of which types exist.
Users can register their custom types when initializing the app or plugin.
```

More concise at the cost of some explicitness.

File: release-content/0.14/release-notes/5781_bevy_reflect_Recursive_registration.md (hunk 1 above)
Comment (MrGVSV) [reply release-content/0.14/release-notes/5781_bevy_reflect_Recursive_registration.md:1]: Do you think we should link to Wikipedia or the [`bevy_reflect` docs](https://docs.rs/bevy_reflect/latest/bevy_reflect/)?

Comment (alice-i-cecile) [reply release-content/0.14/release-notes/5781_bevy_reflect_Recursive_registration.md:1]: Bevy reflect docs please.

Comment (alice-i-cecile) [reply release-content/0.14/release-notes/5781_bevy_reflect_Recursive_registration.md:1]:
```suggestion
Bevy uses [reflection](https://docs.rs/bevy_reflect/latest/bevy_reflect/) in order to dynamically process data for things like serialization and deserialization.
```

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**